### PR TITLE
Test on new Travis CPUs

### PR DESCRIPTION
**Pull request by hugovk**
Helps for #3606.

# New CPU architectures

Now we've moved a lot of testing to GitHub Actions, we should still use Travis CI for testing things which are unique to it, to get good coverage.

Travis has recently made available new CPU architectures, in addition to the existing `amd64`.

* In July, they added `arm64` (Arm support)
  https://blog.travis-ci.com/2019-10-07-multi-cpu-architecture-support

* In November, they added `ppc64le` (IBM Power) and `s390x` (IBM Z CPU)
https://blog.travis-ci.com/2019-11-12-multi-cpu-architecture-ibm-power-ibm-z

* `s390x` is particularly useful as it's big-endian, and we have a number of [big-endian issues](https://github.com/python-pillow/Pillow/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3ABig-endian)

# Changes proposed in this pull request

 * Run tests on the new `arm64`, `ppc64le` and `s390x` architectures

 * Skip install of pyqt5 because "ERROR: Could not find a version that satisfies the requirement pyqt5"

 * The known `s390x` test failures (https://github.com/python-pillow/Pillow/issues/4213) are skipped as expected failures when run on CI: we should still aim to fix them, but there's value in running the other tests

This draft* PR is to illustrate the new CPUs on each Python version. I don't intend we test them all on Python versions, just a single version for each CPU.

(\* Was meant to be a draft but I clicked the wrong button...)

# Cold cache

image

https://travis-ci.org/hugovk/Pillow/builds/632753494

# Warm cache

image

https://travis-ci.org/hugovk/Pillow/builds/632884754

Generally the cold cache build times are about the same for each Python version, with newer versions a bit slower than older ones, except 3.8 on `s390x` is many times slower. 

Warm cache times are closer, but let's optimise for cold caches, which is the case for PR branches.

# Suggestion

Avoid Python 3.8 for now, spread over other versions. For example:

* `s390x` on Python 3.5
* `arm64` on Python 3.6
* `ppc64le` on Python 3.7

### CODE REVIEW

**Comment by wiredfool** [comment]
Oooh. Big Endian.

File: .travis.yml [hunk 1]
```diff
@@ -6,16 +6,23 @@ notifications:
   irc: "chat.freenode.net#pil"
 
 # Run fast lint first to get fast feedback.
-# Run slow PyPy next, to give it a headstart and reduce waiting time.
-# Run latest 3.x next, to get quick compatibility results.
-# Then run the remainder.
+# Run slower CPUs next, to give them a headstart and reduce waiting time.
+# Then run the remainder, with fastest Docker jobs last.
```
Comment (radarhere) [inline .travis.yml]: After #4335, there are no longer any Docker jobs.
```suggestion
# Then run the remainder.
```

File: Tests/helper.py [hunk 2]
```diff
@@ -348,6 +349,10 @@ def on_ci():
     )
 
 
+def is_big_endian():
+    return platform.processor() == "s390x"
```
Comment (homm) [inline Tests/helper.py]: There is a constant:
https://docs.python.org/3/library/sys.html#sys.byteorder

Comment (hugovk) [reply Tests/helper.py]: Thanks, updated!

**Comment by hugovk** [comment]
Also moved the scripts from `.travis/` into `.ci/` to because they're now not solely Travis CI, they're also used by GitHub Actions.

**Comment by radarhere** [comment]
You mention in the initial description that you intended this as a draft PR. Is there still work to be done?

**Comment by hugovk** [comment]
No, ready for review!

**Pull request merged**